prometheus-metrics-model/src/main/java/io/prometheus/metrics/model/registry/ExtendedMultiCollector.java

```diff
@@ -0,0 +1,25 @@
+package io.prometheus.metrics.model.registry;
+
+import io.prometheus.metrics.model.snapshots.MetricSnapshots;
+
+public abstract class ExtendedMultiCollector implements MultiCollector {
```

Comment:
I think this can be removed and users just extend `MultiCollector` directly.

Reply:
What about removing `ExtendedMultiCollector` and adding all methods to the `MultiCollector` interface? Same as you did for `Collector`?

Reply:
Well, it is just to simplify specific implementation. Beware that collectMetricSnapshots is a protected method.
Anyway, default implementation of MultiCollector.collect(PrometheusScrapeRequest scrapeRequest) simply ignores the scrapeRequest, while ExtendeMultiCollector changes this behaviour forcing the implementation of collectMetricSnapshots

Reply:
Thanks @ganzuoni. I understand the argument that by making `ExtendeMultiCollector` enforces an implementation of `collectMetricSnapshots()`. As always in software engineering, it's a trade-off. I think keeping the API simpler (no additional abstract base classes) is the better trade-off in that case.

* It's not clear that returning an empty result is always the best default behavior for `collect()`. For example, in some cases throwing an Exception to trigger an HTTP 500 response might be a good idea, because then the target shows as `failed` in the Prometheus server, and users see directly that there's something wrong. If we stick with the `MultiCollector` interface users must explicitly implement a `collect()` method, and I think this is better than providing a default implementation in the core API.
* `Collector` and `MultiCollector` are the same except for the return type. So, if it is a good idea to have an `ExtendeMultiCollector` in addition to `MultiCollector`, it will also be a good idea to have an `ExtendeCollector` in addition to `Collector`. If we really follow through with this, the core API gets two new base classes. I think it's not worth the complexity for the multi target use case.

Reply:
Agree